### ISSUE TITLE
rvm1:install:gems Could not find rake

### ISSUE THREAD

**Issue by apellizzn** [rvm/rvm1-capistrano3#51]
Hi i am using, rvm1-capistrano3 1.3.2.2 and capistrano 3.3.5,
when i run bundle exec cap production rvm1:install:gems it gives the following error 
```
INFO [6ceb4147] Running /var/www/app/rvm1scripts/rvm-auto.sh . bundle exec gem install --file Gemfile on <ip>
DEBUG [6ceb4147] Command: cd /var/www/app/releases/20150130005851 && /var/www/app/rvm1scripts/rvm-auto.sh . bundle exec gem install --file Gemfile
DEBUG [aa3aabb2] 	/home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/spec_set.rb:92:in `block in materialize'
DEBUG [aa3aabb2] 	: 
DEBUG [aa3aabb2] 	Could not find rake-10.4.2 in any of the sources
DEBUG [aa3aabb2] 	 (
DEBUG [aa3aabb2] 	Bundler::GemNotFound
DEBUG [aa3aabb2] 	)
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/spec_set.rb:85:in `map!'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/spec_set.rb:85:in `materialize'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/definition.rb:133:in `specs'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/definition.rb:178:in `specs_for'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/definition.rb:167:in `requested_specs'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/environment.rb:18:in `requested_specs'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/runtime.rb:13:in `setup'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler.rb:120:in `setup'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/gems/ruby-2.1.2@global/gems/bundler-1.6.2/lib/bundler/setup.rb:17:in `<top (required)>'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
DEBUG [aa3aabb2] 		from /home/ubuntu/.rvm/rubies/ruby-2.1.2/lib/ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
```
 Capfile
```
require 'capistrano/setup'
require 'capistrano/deploy'
require 'rvm1/capistrano3'
require 'capistrano/rails'
Dir.glob('lib/capistrano/tasks/*.rake').each { |r| import r }
```
deploy.rb
```
before 'deploy', 'rvm1:install:rvm'
before 'deploy', 'rvm1:install:ruby'  # install/update Ruby
before 'deploy', 'rvm1:install:gems'  # install/update gems from Gemfile into gemset
```

**Comment by mpapis** [comment]
is your `Capfile` exactly as above? can you also show `lib/capistrano/tasks/*.rake` - [gist it](https://gist.github.com) - can I also see `Gemfile`?

**Comment by apellizzn** [comment]
Yes my ```Capfile``` is exactly the above, my [Gemfile](https://gist.github.com/apellizzn/30c7ff1fdd94def81308), i have no tasks inside /capistrano/tasks folder

**Comment by x-yuri** [comment]
I'm experiencing the same issue.

**Comment by mpapis** [comment]
something is prefixing `gem install` with `bundle exec` - this causes the process to search for `rake` before it is installed, please show full output, maybe you can get more details with `--backtrace` option?

**Comment by x-yuri** [comment]
`./bin/cap vagrant deploy`:

```
...
INFO [9327a64b] Running /home/yuri/rvm1scripts/rvm-auto.sh . bundle exec gem install --file Gemfile as yuri@192.168.88.10
DEBUG [9327a64b] Command: cd /home/yuri/releases/20150223182227 && /home/yuri/rvm1scripts/rvm-auto.sh . bundle exec gem install --file Gemfile
DEBUG [9327a64b]        /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:92:in `block in materialize': Could not find rake-10.4.2 in any of the sources (Bundler::GemNotFound)
DEBUG [9327a64b]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `map!'
DEBUG [9327a64b]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `materialize'
DEBUG [9327a64b]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:132:in `specs'
DEBUG [9327a64b]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:177:in `specs_for'
DEBUG [9327a64b]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:166:in `requested_specs'
DEBUG [9327a64b]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/environment.rb:18:in `requested_specs'
DEBUG [9327a64b]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/runtime.rb:13:in `setup'
DEBUG [9327a64b]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler.rb:122:in `setup'
DEBUG [9327a64b]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/setup.rb:18:in `<top (required)>'
DEBUG [9327a64b]                from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
DEBUG [9327a64b]                from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
(Backtrace restricted to imported tasks)
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as yuri@192.168.88.10: gem exit status: 1
gem stdout: Nothing written
gem stderr: /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:92:in `block in materialize': Could not find rake-10.4.2 in any of the sources (Bundler::GemNotFound)
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `map!'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `materialize'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:132:in `specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:177:in `specs_for'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:166:in `requested_specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/environment.rb:18:in `requested_specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/runtime.rb:13:in `setup'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler.rb:122:in `setup'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/setup.rb:18:in `<top (required)>'
        from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'

SSHKit::Command::Failed: gem exit status: 1
gem stdout: Nothing written
gem stderr: /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:92:in `block in materialize': Could not find rake-10.4.2 in any of the sources (Bundler::GemNotFound)
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `map!'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `materialize'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:132:in `specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:177:in `specs_for'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:166:in `requested_specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/environment.rb:18:in `requested_specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/runtime.rb:13:in `setup'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler.rb:122:in `setup'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/setup.rb:18:in `<top (required)>'
        from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'

Tasks: TOP => deploy => rvm1:install:gems
(See full trace by running task with --trace)
```

`./bin/cap vagrant deploy --backtrace`:

```
...
INFO [cc40b868] Running /home/yuri/rvm1scripts/rvm-auto.sh . bundle exec gem install --file Gemfile as yuri@192.168.88.10
DEBUG [cc40b868] Command: cd /home/yuri/releases/20150223184859 && /home/yuri/rvm1scripts/rvm-auto.sh . bundle exec gem install --file Gemfile
DEBUG [cc40b868]        /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:92:in `block in materialize': Could not find rake-10.4.2 in any of the sources (Bundler::GemNotFound)
DEBUG [cc40b868]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `map!'
DEBUG [cc40b868]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `materialize'
DEBUG [cc40b868]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:132:in `specs'
DEBUG [cc40b868]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:177:in `specs_for'
DEBUG [cc40b868]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:166:in `requested_specs'
DEBUG [cc40b868]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/environment.rb:18:in `requested_specs'
DEBUG [cc40b868]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/runtime.rb:13:in `setup'
DEBUG [cc40b868]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler.rb:122:in `setup'
DEBUG [cc40b868]                from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/setup.rb:18:in `<top (required)>'
DEBUG [cc40b868]                from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
DEBUG [cc40b868]                from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as yuri@192.168.88.10: gem exit status: 1
gem stdout: Nothing written
gem stderr: /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:92:in `block in materialize': Could not find rake-10.4.2 in any of the sources (Bundler::GemNotFound)
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `map!'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `materialize'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:132:in `specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:177:in `specs_for'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:166:in `requested_specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/environment.rb:18:in `requested_specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/runtime.rb:13:in `setup'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler.rb:122:in `setup'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/setup.rb:18:in `<top (required)>'
        from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/command.rb:97:in `exit_status='
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:153:in `block (5 levels) in _execute'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:551:in `call'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:551:in `do_request'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:562:in `channel_request'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:466:in `dispatch_incoming_packets'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:222:in `preprocess'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:206:in `process'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `block in loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:269:in `wait'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:175:in `block (3 levels) in _execute'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:514:in `call'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:514:in `do_open_confirmation'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:546:in `channel_open_confirmation'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:466:in `dispatch_incoming_packets'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:222:in `preprocess'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:206:in `process'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `block in loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `loop'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:177:in `block (2 levels) in _execute'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:191:in `with_ssh'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:136:in `block in _execute'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:133:in `tap'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:133:in `_execute'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:66:in `execute'
/home/yuri/.gem/ruby/2.1.5/gems/rvm1-capistrano3-1.3.2.2/lib/rvm1/tasks/capistrano3/commands.rake:29:in `block (5 levels) in <top (required)>'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/abstract.rb:77:in `within'
/home/yuri/.gem/ruby/2.1.5/gems/rvm1-capistrano3-1.3.2.2/lib/rvm1/tasks/capistrano3/commands.rake:28:in `block (4 levels) in <top (required)>'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:54:in `instance_exec'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:54:in `run'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/runners/parallel.rb:13:in `block (2 levels) in execute'
SSHKit::Command::Failed: gem exit status: 1
gem stdout: Nothing written
gem stderr: /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:92:in `block in materialize': Could not find rake-10.4.2 in any of the sources (Bundler::GemNotFound)
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `map!'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/spec_set.rb:85:in `materialize'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:132:in `specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:177:in `specs_for'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/definition.rb:166:in `requested_specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/environment.rb:18:in `requested_specs'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/runtime.rb:13:in `setup'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler.rb:122:in `setup'
        from /home/yuri/.rvm/gems/ruby-2.1.1@global/gems/bundler-1.8.2/lib/bundler/setup.rb:18:in `<top (required)>'
        from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
        from /home/yuri/.rvm/rubies/ruby-2.1.1/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:54:in `require'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/command.rb:97:in `exit_status='
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:153:in `block (5 levels) in _execute'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:551:in `call'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:551:in `do_request'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:562:in `channel_request'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:466:in `dispatch_incoming_packets'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:222:in `preprocess'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:206:in `process'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `block in loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:269:in `wait'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:175:in `block (3 levels) in _execute'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:514:in `call'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/channel.rb:514:in `do_open_confirmation'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:546:in `channel_open_confirmation'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:466:in `dispatch_incoming_packets'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:222:in `preprocess'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:206:in `process'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `block in loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `loop'
/home/yuri/.gem/ruby/2.1.5/gems/net-ssh-2.9.2/lib/net/ssh/connection/session.rb:170:in `loop'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:177:in `block (2 levels) in _execute'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:191:in `with_ssh'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:136:in `block in _execute'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:133:in `tap'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:133:in `_execute'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:66:in `execute'
/home/yuri/.gem/ruby/2.1.5/gems/rvm1-capistrano3-1.3.2.2/lib/rvm1/tasks/capistrano3/commands.rake:29:in `block (5 levels) in <top (required)>'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/abstract.rb:77:in `within'
/home/yuri/.gem/ruby/2.1.5/gems/rvm1-capistrano3-1.3.2.2/lib/rvm1/tasks/capistrano3/commands.rake:28:in `block (4 levels) in <top (required)>'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:54:in `instance_exec'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/backends/netssh.rb:54:in `run'
/home/yuri/.gem/ruby/2.1.5/gems/sshkit-1.6.1/lib/sshkit/runners/parallel.rb:13:in `block (2 levels) in execute'
Tasks: TOP => deploy => rvm1:install:gems
```

I'm deploying to a test VM, if anything.

**Comment by x-yuri** [comment]
> something is prefixing gem install with bundle exec

Not so surprisingly, it's... `capistrano-bundler` :) It wasn't me who decided to use gemsets with `bundler` together in one project.

**Comment by vokshirg** [comment]
You can fix it with task: 
```
  task :install_bundler do
    on roles :all do
      execute :rvm, "all do gem install bundler"
    end
  end
  after "rvm1:install:rvm", "deploy:install_bundler"
```

**Comment by JohnSmall** [comment]
> You can fix it with task:
> 
> ```
>   task :install_bundler do
>     on roles :all do
>       execute :rvm, "all do gem install bundler"
>     end
>   end
>   after "rvm1:install:rvm", "deploy:install_bundler"
> ```

Doesn't work for me. It installs bundler OK, but it still complains about rake being missing. I get

` deploy:install_bundler
      01 /var/www/staging-indra/rvm1scripts/rvm-auto.sh rvm all do gem install bundler
      01 Successfully installed bundler-1.17.1
      01 Parsing documentation for bundler-1.17.1
      01 Done installing documentation for bundler after 2 seconds
      01 1 gem installed
    ✔ 01 john@my_machine 4.572s
00:16 deploy:install_rake
      01 /var/www/staging-indra/rvm1scripts/rvm-auto.sh rvm all do gem pristine gem-wrappers --version 1.4.0
      01 Restoring gems to pristine condition...
      01 Building native extensions. This could take a while...
      01 Restored gem-wrappers-1.4.0
    ✔ 01 john@my_machine 0.660s
      02 /var/www/staging-indra/rvm1scripts/rvm-auto.sh rvm all do gem install rake
      02 Successfully installed rake-12.3.1
      02 Parsing documentation for rake-12.3.1
      02 Done installing documentation for rake after 0 seconds
      02 1 gem installed
    ✔ 02 john@my_machine 1.951s
00:19 rvm1:install:gems
      01 /var/www/staging-indra/rvm1scripts/rvm-auto.sh . bundle exec gem install --file Gemfile
      01 Ignoring gem-wrappers-1.4.0 because its extensions are not built. Try: gem pristine gem-wrappers --version 1.4.0
      01 bundler: failed to load command: gem (/home/john/.rvm/rubies/ruby-2.5.1/bin/gem)
      01 Bundler::GemNotFound: Could not find rake-12.3.1 in any of the sources
`

Basically I see the output which tells me XYZ is not loaded, so I add a new task to load what's missing and try again, and then get exactly the same error.

As you can see from the output, it installs  rake-12.3.1 and then complains that it's missing.

In the past I've just shrugged my shoulders, ssh'd into the machine, installed the missing items manually and carried on using Capistrano. But now I really want to be able to install the whole thing from scratch on a new machine using only Capistrano, and I can't do it.

Should I remove `capistrano/bundler` from my Capfile ?
